cmd/common/image/build.go

```diff
@@ -17,80 +17,202 @@ package image
+type cmdOpts struct {
+	path         string
+	file         string
+	fileChanged  bool
+	image        string
+	local        bool
+	builderImage string
 }
 
 func NewBuildCmd() *cobra.Command {
+	opts := &cmdOpts{}
+
 	cmd := &cobra.Command{
-		Use:          "build IMAGE",
+		Use:          "build PATH",
 		Short:        "Build a gadget image",
 		SilenceUsage: true,
-		RunE:         runBuild,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("expected exactly one unnamed argument")
+			}
+
+			if opts.local && opts.builderImage != DEFAULT_BUILDER_IMAGE {
+				return fmt.Errorf("--local and --builder-image cannot be used at the same time")
+			}
+
+			fFlag := cmd.Flags().Lookup("file")
+			opts.fileChanged = fFlag.Changed
+
+			opts.path = args[0]
+
+			return runBuild(opts)
+		},
 	}
 
-	cmd.Flags().String("prog-amd64", "", "path to the amd64 variant of the eBPF program")
-	cmd.Flags().String("prog-arm64", "", "path to the arm64 variant of the eBPF program")
-	cmd.Flags().String("definition", "", "path to the definition file")
+	cmd.Flags().StringVarP(&opts.file, "file", "f", "build.yaml", "path to build.yaml")
+	cmd.Flags().BoolVarP(&opts.local, "local", "l", false, "build using local tools")
+	cmd.Flags().StringVarP(&opts.image, "tag", "t", "", "name for the built image (format name:tag)")
+	cmd.Flags().StringVar(&opts.builderImage, "builder-image", DEFAULT_BUILDER_IMAGE, "builder image to use")
 
 	return cmd
 }
 
-func runBuild(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 {
-		return fmt.Errorf("expected exactly one unnamed argument")
+type imageIndexOpts struct {
+	progAmd64FilePath  string
+	progArm64FilePath  string
+	definitionFilePath string
+	image              string
+}
+
+func runBuild(opts *cmdOpts) error {
+	conf := &buildFile{
+		EBPFProgram: "program.bpf.c",
+		Definition:  "definition.yaml",
 	}
-	o := &buildOptions{
-		progAmd64FilePath:  cmd.Flag("prog-amd64").Value.String(),
-		progArm64FilePath:  cmd.Flag("prog-arm64").Value.String(),
-		definitionFilePath: cmd.Flag("definition").Value.String(),
-		image:              args[0],
+
+	var buildContent []byte
+	var err error
+
+	if opts.fileChanged {
+		buildContent, err = os.ReadFile(opts.file)
+		if err != nil {
+			return fmt.Errorf("reading build file: %w", err)
+		}
+	} else {
+		buildContent, err = os.ReadFile(filepath.Join(opts.path, opts.file))
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("reading build file: %w", err)
+		}
 	}
 
-	if o.progArm64FilePath == "" && o.progAmd64FilePath == "" {
-		return fmt.Errorf("at least one of --prog-amd64 or --prog-arm64 must be specified")
+	if len(buildContent) > 0 {
+		if err := yaml.Unmarshal(buildContent, conf); err != nil {
+			return fmt.Errorf("unmarshal build.yaml: %w", err)
+		}
 	}
 
-	ociStore, err := utils.GetLocalOciStore()
+	// make a temp folder to store the build results
+	tmpDir, err := os.MkdirTemp("", "gadget-build")
```

Comment:
Separate the name from the random suffix:

```suggestion
	tmpDir, err := os.MkdirTemp("", "gadget-build-")
```